File: app/src/main/java/quickus/core/QuickusContext.java
package quickus.core;

import java.lang.reflect.AnnotatedType;
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class QuickusContext {
    protected Map<String, Object> parsedParams = new HashMap<>();

    public void clearStepContext() {
        parsedParams.clear();
    }

    public void setParameter(String key, Object value) {
        if (parsedParams.containsKey(key)) {
            System.out.println("WARN: The step already contains a param named [" + key + "]");
        }
        parsedParams.put(key, value);
    }

    public <T> T getParameter(String parameter, Class<T> clazz) {
        return clazz.cast(parsedParams.get(parameter));
    }
}

File: src/main/java/quickus/core/QuickusContext.kt
package quickus.core

import java.util.HashMap

class QuickusContext {
    protected var parsedParams: MutableMap<String, Any> = HashMap()
    fun clearStepContext() {
        parsedParams.clear()
    }

    fun setParameter(key: String, value: Any) {
        if (parsedParams.containsKey(key)) {
            println("WARN: The step already contains a param named [$key]")
        }
        parsedParams[key] = value
    }

    fun <T> getParameter(parameter: String, clazz: Class<T>): T {
        return clazz.cast(parsedParams[parameter])
    }
}